src/components/Card.js

```diff
@@ -1,62 +1,90 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleCardImage) {
+  constructor( {
+    name, 
+    link,  
+    isLiked,
+    _id,
+  },
+    cardSelector,
+    handleCardImage,
+    handleDelete,
+    userId,
+    handleCardLike,
+
+  ) {
     this._name = name;
     this._link = link;
+    this._isLiked = isLiked;
+    this._cardId = _id;
     this._cardSelector = cardSelector;
     this._handleCardImage = handleCardImage;
+    this._handleDeleteCard = handleDelete;
+    this._userId = userId;
+    this._handleCardLike = handleCardLike
+
+  }
+
+  getView() {
+    this._cardElement = this._getTemplate();
+    this._handleCardDeleteButton = this._cardElement.querySelector(".card__delete-button");
+    this._cardLikeButton = this._cardElement.querySelector(".card__like-button");
+    this._setEventListeners();
+    this.renderLikes();
+    this._cardElement.querySelector(".card__like-button");
+    this._cardElement.querySelector(".card__image").src = this._link;
+    this._cardElement.querySelector(".card__image").alt = this._name;
+    this._cardElement.querySelector(".card__title").textContent = this._name;
+    return this._cardElement;
   }
 
+  renderLikes() {
+    if (this._isLiked ) {
+      this._cardLikeButton.classList.add("card__like-button_active");
+    } else {
+      this._cardLikeButton.classList.remove("card__like-button_active");
+    }
+  }
 
-  //_handleCardImage = (e) => {
-    //imagePreview.src = this._link;
-    //imagePreview.alt = this._name;
-    //imgPreviewTitle.textContent = this._name;
-  //};
+  updateIsLiked(isLiked) {
+    this._isLiked  = isLiked;
+    this.renderLikes();
+  }
 
-  _handleDeleteCard() {
+  handleDeleteCard() {
     this._cardElement.remove();
-    this._cardElement = null;
   }
 
+
   _handleLikeButton() {
     this._cardElement
       .querySelector(`.card__like-button`)

```

Comment:
`    this._cardElement.querySelector(".card__like-button");
`

The like button has been found in line 30 as `this._cardLikeButton`. Please, use it here


```js

  _handleLikeButton() {
    this._cardLikeButton.classList.toggle(`card__like-button_active`);
  }
```